src/charm.py

```diff
@@ -142,6 +133,21 @@ def generate_worker_layer(self, worker: Worker) -> Layer:
                         "summary": "tempo worker process",
                         "command": f"/bin/tempo -config.file={CONFIG_FILE} -target {role}",
                         "startup": "enabled",
+                        # Configure Tempo workload traces
+                        "environment": {
```

Comment:
rather than unconditionally pass an environment with potentially half-initialized variables, I'd rather build the environment apart and then pass it in. E.g.

```python
env = {}
if tempo_endpoint := worker.cluster...
    otel_env = {OTEL_EXPORTER_JAEGER...
    env.update(otel_env)
    
return Layer(... environment=env)
```

Reply:
(reason is: you never know what a binary will do if it sees OTEL_RESOURCE_ATTRIBUTES but no OTEL_EXPORTER_JAEGER_ENDPOINT for example... better keep the env clean)